adapters/ui-embeddable/components/client.tsx

```diff
@@ -0,0 +1,30 @@
+import { initClient } from "@ts-rest/core";
+import { contract } from "./contract";
+import { useEffect, useState } from "react";
+
+const isServer = typeof window === "undefined";
+
+const getBaseUrl = () => {
+  if (isServer) {
+    return `${process.env.NEXT_PUBLIC_INFERABLE_API_URL || "https://api.inferable.ai"}`;
+  }
+
+  return `/api`;
+};
+
+const client = (customAuthToken: string) => initClient(contract, {
+  baseUrl: getBaseUrl(),
+  baseHeaders: {
+    authorization: `custom ${customAuthToken}`,
+  },
+});
+
+export const useClient = (customAuthToken: string) => {
```

Comment:
Is this used?

Are there endpoints we need access to that aren't available through `useRun`?